### PR TITLE
Fix issue in updating the GraphQL API and updating subscription policy

### PR DESCRIPTION
**Pull request by vithu30**
Fixes https://github.com/wso2/product-apim/issues/9568 and fixes https://github.com/wso2/product-apim/issues/10940 and fixes https://github.com/wso2/product-apim/issues/10678

Git Actions link - https://github.com/vithu30/carbon-apimgt-1/actions/runs/737974733

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java [hunk 1]
```diff
@@ -108,16 +108,16 @@ public static SubscriptionThrottlePolicyDTO fromSubscriptionThrottlePolicyToDTO(
...
-            MonetizationInfoDTO monetizationInfoDTO = new MonetizationInfoDTO();
             if (APIConstants.Monetization.FIXED_RATE.equalsIgnoreCase(subscriptionPolicy.getMonetizationPlan())) {
                 monetizationInfoDTO.setMonetizationPlan(MonetizationInfoDTO.MonetizationPlanEnum.FIXEDRATE);
             } else {
                 monetizationInfoDTO.setMonetizationPlan(MonetizationInfoDTO.MonetizationPlanEnum.DYNAMICRATE);
             }
             monetizationInfoDTO.setProperties(subscriptionPolicy.getMonetizationPlanProperties());
-            policyDTO.setMonetization(monetizationInfoDTO);
         }
+        policyDTO.setMonetization(monetizationInfoDTO);
```
Comment (shilmyhasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java:120]: Do we need to set this , if the subscription policy is not a commercial one?

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx [hunk 2]
```diff
@@ -237,7 +237,7 @@ function AddEdit(props) {
                 setValidRoles(result.body.permissions
                     && result.body.permissions.roles
                     ? result.body.permissions.roles
-                    : []);
+                    : ['Internal/everyone']);
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx:240]: Is it really needed to add it? we can have subscription policies without Internal/everyone permission..

Comment (vithu30) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx:240]: In the REST API definition, the parameter `roles` is marked as required one - [1]. We should either send a default value or mark it as a not required parameter.In earlier versions, also we are sending this default value. 
Then shall we mark it as a not required parameter. WDYT?

[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml#L3256

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java (hunk 1 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java:120]: Yes, the `monetizationPlan` is marked as a required attribute and it is an enum - [1]. Currently, it returns null if `COMMERCIAL_TIER_PLAN` is not selected - [2] and throws NPE when editing subscription policy [3]. Earlier this issue was not there as if the `monetizationPlan` is null then UI sent `FIXEDRATE`. This logic is changed in a recent fix.

So in order to fix it, either we should mark it as a not-required value or set a default value. Then shall we mark it as a not-required attribute. WDYT?

[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/resources/admin-api.yaml#L3522 
[2] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java#L111
[3] https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx#L258

Comment (shilmyhasan) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java:120]: +1

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx (hunk 2 above)
Comment (malinthaprasan) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx:240]: I think we can remove it as a required one.. 
@chamilaa did a similar fix recently..

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java (hunk 1 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/SubscriptionThrottlePolicyMappingUtil.java]: Fixed in db16b57

File: features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx (hunk 2 above)
Comment (vithu30) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx]: +1. Yes, we can send null if permission is not required for a policy. Handled it in UI by adding a new option `None`:

image

Comment (vithu30) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx]: Fixed

Comment (malinthaprasan) [reply features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Throttling/Subscription/AddEdit.jsx]: Great!

**Review by malinthaprasan — APPROVED** [review]

**Comment by malinthaprasan** [comment]
@tmkasun please review the react section and merge

**Review by tmkasun — APPROVED** [review]

**Pull request merged**